src/main/java/frc/robot/subsystems/IntakeSubsystem.java

```diff
@@ -71,13 +87,54 @@ public void close() throws Exception {
     // Mostly for JUnit tests
     motor.close();
     motor2.close();
-    
+  }
+
+  public double getConeDist() {
+    return timeOfFlightSensor.getRawDistance0().classDistance;
+  }
+
+  public double getCubeDist() {
+    return timeOfFlightSensor.getRawDistance1().classDistance;
+  }
+
+  public void changeFlightState() {
+    double coneDistance = getConeDist();
+    double cubeDistance = getCubeDist();
+
+    boolean coneSensorUp = coneDistance != -1;
+    boolean cubeSensorUp = cubeDistance != -1;
+
+    SmartDashboard.putNumber("Cone Distance", coneDistance);
+    SmartDashboard.putNumber("Cube Distance", cubeDistance);
+    SmartDashboard.putBoolean("Has Cone", coneDistance <= coneActivationThreshold);
+    SmartDashboard.putBoolean("Has Cube", cubeDistance <= cubeActivationThreshold);
+    SmartDashboard.putBoolean("Cone Sensor Working", coneSensorUp);
+    SmartDashboard.putBoolean("Cube Sensor Working", cubeSensorUp);
+
+    if (coneSensorUp && cubeSensorUp) {
+      // Only change state if both sensors are up, otherwise stay in last state
+      if (coneDistance <= coneActivationThreshold) {
+        tofState = FlightStates.CONE;
+      } 
+      else if (cubeDistance <= cubeActivationThreshold) {
+        tofState = FlightStates.CUBE;
+      } 
+      else {
+        tofState = FlightStates.IDLE;
+      }
+    }
+  }
+
+  public FlightStates getFlightState() {
+    // Check if state has changed before returning it
+    changeFlightState();
+    return tofState;
   }
 
   @Override
   public void periodic() {
-    intakeMode.setString((isConeMode) ? "Cone Mode" : "Cube Mode");
     // This method will be called once per scheduler run
+    intakeMode.setString((isConeMode) ? "Cone Mode" : "Cube Mode");
```

Comment:
Why in periodic?  Let's make this update only when it changes.

Reply:
Moved to setMode method. Should only update when changed